core/Execute.cpp

```diff
@@ -1,47 +1,195 @@
+                std::unique_ptr<sparta::ResourceTreeNode> exepipe =
+                    std::make_unique<sparta::ResourceTreeNode>(
+                        unit_name, tgt_name, pipeidx, std::string(unit_name + " Execution Pipe"),
+                        &exe_pipe_fact_);
+                exepipe->addAlias(exe_pipe_alias[pipeidx][1]);
+                node->addChild(*exepipe);
+                exe_pipe_tns_.emplace_back(std::move(exepipe));
+            }
+            else
+            {
+                // no alias
+                exe_pipe_tns_.emplace_back(new sparta::ResourceTreeNode(
+                    node, unit_name, tgt_name, pipeidx, std::string(unit_name + " Execution Pipe"),
+                    &exe_pipe_fact_));
             }
         }
     }
 
-    void ExecuteFactory::deleteSubtree(sparta::ResourceTreeNode*) {
+    void ExecuteFactory::bindLate(sparta::TreeNode* node)
+    {
+        /*
+            For issue queue we need to establish mappings such that a mapping of target pipe to
+           execution pipe in an issue queue is known such as:
+                    iq_0:
+                    "int": exe0, exe1
+                    "div": exe1
+                    "mul": exe2
+            so when we have an instruction, we can get the target pipe of an instruction and lookup
+           available execution units
+        */
+
+        auto pipelines = olympia::coreutils::getPipeTopology(node->getParent(), "pipelines");
+        std::unordered_map<std::string, int> exe_pipe_to_iq_number;
+
+        for (size_t iq_num = 0; iq_num < issue_queue_to_pipe_map_.size(); ++iq_num)
+        {
+            auto iq = issue_queue_to_pipe_map_[iq_num];
+            auto pipe_target_start = std::stoi(iq[0]);
+            auto pipe_target_end = pipe_target_start;
+            if (iq.size() > 1)
+            {
+                pipe_target_end = std::stoi(iq[1]);
+            }
+            pipe_target_end++;
+            for (int pipe_idx = pipe_target_start; pipe_idx < pipe_target_end; ++pipe_idx)
+            {
+                const std::string exe_name = "exe" + std::to_string(pipe_idx);
+                for (const auto & exe_pipe_tns : exe_pipe_tns_)
+                {
+                    olympia::ExecutePipe* exe_pipe =
+                        exe_pipe_tns->getResourceAs<olympia::ExecutePipe*>();
+                    const std::string exe_pipe_name = exe_pipe_tns->getName();
+                    // check if the names match, then we have the correct exe_pipe_
+                    if (exe_name == exe_pipe_name)
+                    {
+                        auto pipe_targets = pipelines[pipe_idx];
+                        // if execution has branch pipe target
+                        if (std::find(pipe_targets.begin(), pipe_targets.end(), "br")
+                            != pipe_targets.end())
+                        {
+                            exe_pipe->setBranchRandomMisprediction(true);
```

Comment:
I remember our talking about this, but I think I there's some  miscommunication.  You NEVER want to force an override of a parameter this way -- this is confusing to the user/modeler.

I have a patch for this -- the better way to do this.  I'll attached it to this PR.

Reply:
The patch to Execute.cpp:

[Execute.cpp.patch.txt](https://github.com/riscv-software-src/riscv-perf-model/files/14169603/Execute.cpp.patch.txt)

The changes to ExecutionPipe.hpp/cpp:
```
/// header file/parameters:
            PARAMETER(bool, enable_random_misprediction, false,
                      "test mode to inject random branch mispredictions")
            HIDDEN_PARAMETER(bool, contains_branch_unit, false,
                             "Does this exe pipe contain a branch unit")

        const bool     enable_random_misprediction_;

// source file:

        enable_random_misprediction_(p->contains_branch_unit && p->enable_random_misprediction),


        // Ignore if this pipe does not have a branch unit
        p->enable_random_misprediction.ignore();

```

Reply:
Patch implemented in [78e5ba0](https://github.com/riscv-software-src/riscv-perf-model/pull/140/commits/78e5ba04f059d1dfb22987c6aca4a66ce6069859)